### PR TITLE
Allow Invisibility and Aussie Ping To Affect All Attacks Created By Hermits

### PR DESCRIPTION
**Pull request by Lunarmagpie**
Attempts to fix the consistency issues between Invisibility and Aussie Ping noted by Jamm.

> so because of this,  i've been playing around with Invis which I don't use and I noticed that Invis
> - can double/block lLamadad's additional damage (+40)
> - can double/block Goatfather's +30 but not the +10 to hermits below the active
> - can't double/block any of the Skizzs additional damage
> - can't double/block S. Stress +10 to AFK
>
> compared to Pearl
> - Pearl can block lLamadad's additional damage (+40)
> - Pearl can block Goatfather's +30 and +10 to hermits below the active
> - Pearl can't block any of Skizzs additional damage
> - Pearl can block S. Stress +10 to AFK

This PR makes Invisibility Potion and Pearl trigger in all of these scenarios.

Add tests for both regular skizz''s to ensure that the tests with pearl and invisibility are not relying on incorrect behavior.

### CODE REVIEW

File: common/status-effects/invisibility-potion.ts [hunk 1]
```diff
@@ -21,10 +21,16 @@ export const InvisibilityPotionHeadsEffect: StatusEffect<PlayerComponent> = {
...
 			(attack) => {
 				if (!attack.isType('primary', 'secondary')) return
 				attack.multiplyDamage(effect.entity, 0)
+			},
+		)
+		observer.subscribeWithPriority(
+			player.opponentPlayer.hooks.onTurnEnd,
+			onTurnEnd.ON_STATUS_EFFECT_TIMEOUT,
+			() => {
 				effect.remove()
```
Comment (Screaper91) [inline common/status-effects/invisibility-potion.ts:34]: Invisibility status effects should be persistent until they modify an attack
```suggestion
				if (multipliedDamage) effect.remove()
```

File: tests/game/invisibility-potion.test.ts [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+							query.row.opponentPlayer,
+						)?.health,
+					).toBe(EthosLabCommon.health - 30 /* Anvil */)
+					expect(
+						game.components.find(
+							RowComponent,
+							query.row.index(1),
+							query.row.opponentPlayer,
+						)?.health,
+					).toBe(EthosLabCommon.health - 10 /* Anvil */)
```
Comment (Screaper91) [inline tests/game/invisibility-potion.test.ts:112]: Modify this test to check AFK Etho's health after playerTwo ends their turn
```suggestion
					yield* endTurn(game)
					expect(
						game.components.find(
							RowComponent,
							query.row.index(1),
							query.row.currentPlayer,
						)?.health,
					).toBe(EthosLabCommon.health - 10 /* Anvil only */)
```

**Review by Screaper91 — CHANGES_REQUESTED** [review]
This looks good, I'll share additional `onTurnEnd` tests for other cards affected by these changes before this is merged.

File: common/status-effects/invisibility-potion.ts (hunk 1 above)
Comment (Lunarmagpie) [reply common/status-effects/invisibility-potion.ts:34]: Good catch! I didn't realize it should last multiple turns.

**Comment by Lunarmagpie** [comment]
> This looks good, I'll share additional `onTurnEnd` tests for other cards affected by these changes before this is merged.

I sent you a collaborator invite for my fork if you would like to write them to my branch.

**Pull request merged**